### ISSUE TITLE
Speech Note crashes after start on Ubuntu 23.10

### ISSUE THREAD

**Issue by carroll1955** [mkiol/dsnote#86]
Starting from the icon the application runs but crashes *very* shortly after opening.

I have 

## Hardware Information:
- **Hardware Model:**                              (null)
- **Memory:**                                      4.0 GiB
- **Processor:**                                   AMD Athlon™ II X2 220 × 2
- **Graphics:**                                    AMD RS780
- **Disk Capacity:**                               360.1 GB

## Software Information:
- **Firmware Version:**                            6.01
- **OS Name:**                                     Ubuntu 23.10
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.1
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.5.0-14-generic

I am running Ubuntu 23.10

> flatpak run net.mkiol.SpeechNote --verbose
 produces
[speechnote.txt](https://github.com/mkiol/dsnote/files/13864440/speechnote.txt)

I do want to try this out and I would appreciate you help.

Thank you

**Comment by mkiol** [comment]
Thanks for the report.

Could you please try the following:

1. start with `--gpu-scan-off` (disable GPU discovery)

```
flatpak run net.mkiol.SpeechNote --verbose --gpu-scan-off
```

2. if 1) doesn't help, run with `--py-scan-off` (disable python libraries)

```
flatpak run net.mkiol.SpeechNote --verbose --py-scan-off
```

**Comment by carroll1955** [comment]
"
flatpak run net.mkiol.SpeechNote --verbose --py-scan-off
"

Worked and the GUI is now functional

Thank you very much

**Comment by mkiol** [comment]
Glad it helped :)

The problem is that with `--py-scan-off` you have disabled all python-dependent fatures including `Faster Whisper`, `Coqui TTS`, `Mimic 3` models.

If it is not a problem, I would like ask you to help me with debugging what exactly causes the crash. To do so, would you be able to install `beta` version of this app? You can do it by enabling `flathub-beta` channel:

```
flatpak remote-add --user --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
```

uninstall already installed version:

```
flatpak uninstall --delete-data net.mkiol.SpeechNote 
```

install `beta` version:

```
flatpak install --user flathub-beta net.mkiol.SpeechNote
```

run the app:

```
flatpak run net.mkiol.SpeechNote --verbose
```

Can you run the application?

**Comment by carroll1955** [comment]
Not a problem to do as you asked

The install failed see attached
[speechnote.txt](https://github.com/mkiol/dsnote/files/13874899/speechnote.txt)

Your work and your help is appreciated

**Comment by mkiol** [comment]
Can I have one more request? 

Would you be able to start the app with debugger. Follow these commands:

```
flatpak run --command=sh --devel net.mkiol.SpeechNote
```

```
[📦 net.mkiol.SpeechNote ~]$ gdb --args dsnote --verbose
```

```
Enable debuginfod for this session? (y or [n]) y
```

```
(gdb) run
```

wait for the crash


```
(gdb) bt
```

```
(gdb) quit
```

```
[📦 net.mkiol.SpeechNote ~]$ exit
```

Collect the logs and attach them.

I would be very grateful for your help.

**Comment by carroll1955** [comment]
Of course - not a problem

I ran the first command and got 
"
error: runtime/org.kde.Sdk/x86_64/5.15-22.08 not installed
"

Please guide me past that

**Comment by mkiol** [comment]
Oh, SDK runtime is missing. Please install with:

```
flatpak install runtime/org.kde.Sdk/x86_64/5.15-22.08
```

After the test, you can uninstall with:

```
flatpak uninstall runtime/org.kde.Sdk/x86_64/5.15-22.08 
```

**Comment by carroll1955** [comment]
"Collect the logs and attach them."

Forgive my ignorance, but - where do I find the logs ?

**Comment by mkiol** [comment]
> Forgive my ignorance, but - where do I find the logs ?

Sorry. I wasn't very specific. Saying "logs" I meant everything that is on standard output. Everything printed to terminal after `(gdb) run` and `(gdb) bt`.

**Comment by carroll1955** [comment]
Thank you

I hope this is enough / all
[collected.txt](https://github.com/mkiol/dsnote/files/13907032/collected.txt)

**Comment by mkiol** [comment]
I got everything what is needed!

The conclusions are as follows:
- app crashes with "Illegal Instruction" error when PyTorch library is loaded
- PyTorch (shipped together with flatpak package) requires at least `AVX` CPU extension, but your processes is quite old and does not support it - this is a main cause of the crash

Temporary workaround:
- start the app with `--py-scan-off` to disable all python libraries (you need to do it only once)

Long-term fix:
- To prevent crashing, I need to add extra check if CPU has `AVX` extension before loading PyTorch library

Thank you for very much for your help.

(Let's keep this issue open. I will close it when "long-term fix" is ready)

**Comment by carroll1955** [comment]
It runs *very* slowly -- It is indeed an old machine and not up to the task.

Thank you again

**Comment by mkiol** [comment]
Workaround fix: 8cbc56203fc96eed8957ece8bb938cd5c9871b3b (disable PyTorch when CPU don't support AVX)
